packages/wonder-stuff-server/src/types.js

```diff
@@ -0,0 +1,122 @@
+// @flow
+import type {
+    NpmLogLevels,
+    Logger as WinstonLogger,
+    Info as WinstonInfo,
+    Transport,
+} from "winston";
+
+import type {$Request} from "express";
+
+/**
+ * Describes logging metdata.
+ */
+export type Info = WinstonInfo<NpmLogLevels>;
+
+/**
+ * Defines the different log levels.
+ */
+export type LogLevel = $Keys<NpmLogLevels>;
+
+/**
+ * Describes the interface for logging gateway activity.
+ */
+export type Logger = WinstonLogger<NpmLogLevels>;
+
+/**
+ * The runtime modes that a gateway can run under.
+ */
+export enum Runtime {
+    // TODO(somewhatabstract, FEI-4172): Update eslint-plugin-flowtype when
+    // they've fixed https://github.com/gajus/eslint-plugin-flowtype/issues/502
+    /* eslint-disable no-undef */
+    Production = "production",
+    Development = "development",
+    Test = "test",
+    /* eslint-enable no-undef */
+}
+
+/**
+ * Options to configure logging.
+ */
+export type LoggingOptions = {
+    /**
+     * The runtime mode to be targeted. When omitted, this will default to
+     * the current runtime mode as determined from `getRuntimeMode`.
+     */
+    mode?: Runtime,
+
+    /**
+     * Log only if the level of a logged entry is less than or equal to this
+     * level. Enables filtering out of debug message in production, for example.
+     */
+    level: LogLevel,
+
+    /**
+     * Metadata to attach to every logged entry, by default.
+     * Defaults to null.
+     */
+    defaultMetadata?: {...} | (() => {...}),
+
+    /**
+     * The transport to use for handling log entries. When not specified,
+     * this will be determined using `getLoggingTransport`.
+     */
+    transport?: Transport,
```

Comment:
New to allow uses to inject a transport in prod, for example (such as the Google Cloud transport for winston)